src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,279 @@
+    super.disconnectedCallback();
+
+    if (this._unsubEvents) {
+      this._unsubEvents();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this._config.title}">
+        <div class="fabDiv">
+          <paper-fab
+            icon="hass:plus"
+            @click=${this._addItem}
+          ></paper-fab>
+          <paper-fab
+            mini
+            class="moreOptions"
+            icon="hass:notification-clear-all"
+            @click=${this._clearItems}
+          ></paper-fab>
+          <paper-fab
+            mini
+            class="moreOptions"
+            icon="hass:check-all"
+          ></paper-fab>
+        </div>
+        <!--<div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='${this.localize("ui.card.shopping-list.add_item")}'
+            @keydown=${this._addKeyPress}
+            no-label-float
+          ></paper-input>
+        </div>-->
+
+      ${repeat(
+        this._items!,
+        (item) => html`
+        <div class="editRow">
+          <paper-icon-button
+            class="editOption"
+            slot="item-icon"
+            .itemId=${item.id}
+            icon="hass:drag-vertical"
+          ></paper-icon-button>
+          <paper-checkbox
+            slot="item-icon"
+            ?checked=${item.complete}
+            .itemId=${item.id}
+            @click=${this._completeItem}
+            tabindex='0'
+          ></paper-checkbox>
+          <paper-item-body>
+            <paper-input
+              no-label-float
+              .value='${item.name}'
+              .itemId=${item.id}
+              @change=${this._saveEdit}
+            ></paper-input>
+          </paper-item-body>
+          <paper-icon-button
+            class="editOption"
+            slot="item-icon"
+            .itemId=${item.id}
+            icon="hass:close"
+            @click=${this._deleteItem}
+          ></paper-icon-button>
+        </div>
+        `
+      )}
+      </ha-card>
+    `;
+  }
+
+  private renderStyle(): TemplateResult {
+    return html`
+      <style>
+        ha-card {
+          min-height: 180px;
+        }
+        .addRow, .editRow {
+          display: flex;
+          flex-direction: row;
+        }
+        .addRow {
+          padding-left: 16px;
+        }
+        .editOption {
+            visibility: hidden;
+        }
+        .editRow:hover > .editOption {
+            visibility: visible;
+        }
+        .fabDiv {
+          float: right;
+          margin-right: 20px;
+          margin-top: -60px;
+        }
+        paper-fab {
+          display: block;
+          position: relative;
+        }
+        .moreOptions {
+          visibility: hidden;
+          margin-top: 7px;
+          margin-left: 7px;
+        }
+        .fabDiv:hover > .moreOptions {
+          visibility: visible;

```

Comment:
you almost never want to use `visibility` because the elements will keep occupying the space, they are just not visible. You can see this with the delete buttons on your card:

![image](https://user-images.githubusercontent.com/1444314/47620782-a29bef00-daee-11e8-8c77-02398ba203c1.png)

`display: none` is the correct CSS modifier.